packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -184,7 +184,12 @@ export class IncomingRequestsController extends ConsumptionBaseController {
 
         const relationship = await this.relationshipResolver.getRelationshipToIdentity(request.peer);
         // It is safe to decide an incoming Request when no Relationship is found as this is the case when the Request origins from onNewRelationship of the RelationshipTemplateContent
-        if (relationship && relationship.status !== RelationshipStatus.Active) {
+        const possibleStatuses =
+            request.source?.type === "RelationshipTemplate"
+                ? [RelationshipStatus.Active, RelationshipStatus.Rejected, RelationshipStatus.Revoked, RelationshipStatus.Terminated]
+                : [RelationshipStatus.Active];
+
```

Comment:
why should deciding relationship template requests still be allowed if the relationship is inactive?

Reply:
I think terminated could be wrong here. But rejected and revoked are like no relationship exists.

Reply:
Agreed that only Terminated is wrong. Is there a way (planned) to clean up rejected/revoked relationships?

Reply:
I mean, we could use your decompose logic for that. But please let US investigate that further before implementing sth like that.

Reply:
Alright, removed Terminated, should be fine now.